ui/src/shared/components/GetResources.tsx

```diff
@@ -166,11 +173,34 @@ class GetResources extends PureComponent<Props, StateProps> {
   }
 
   public render() {
-    const {resource, children} = this.props
+    const {resources, children} = this.props
+    let status
+    for (let i = 0; i < resources.length; i++) {
+      const resource = resources[i]
+      // reduce the values of the status to 1 value
```

Comment:
Why not use the declarative `Array.prototype.reduce` and put this logic in a getter?

Reply:
When i refactored this to put the logic in a getter I was getting a type error thrown for status being a string (and refactoring the value to return `this.props[resource].status` for each of the cases seemed a bit repetitive). 

As for using reduce, I'd love to talk about how to integrate it without overriding the status value & breaking out of a reduce